gadget/update.go

```diff
@@ -624,6 +672,55 @@ func EnsureVolumeCompatibility(gadgetVolume *Volume, diskVolume *OnDiskVolume, o
 	return gadgetStructIdxToOnDiskStruct, nil
 }
 
+func diskTraitsFromEMMCDevice(dev string, vol *Volume) (res DiskVolumeDeviceTraits, err error) {
+	// Retrieve the primary disk layout of the emmc block device
+	diskLayout, err := OnDiskVolumeFromDevice(dev)
+	if err != nil {
+		return res, fmt.Errorf("cannot read %v partitions for candidate volume %s: %v", dev, vol.Name, err)
+	}
+
+	// ensure emmc is compatible with the gadget emmc setup
+	if _, err := ensureVolumeEMMCCompatibility(vol, diskLayout); err != nil {
+		return res, fmt.Errorf("volume %s is not compatible with disk %s: %v", vol.Name, dev, err)
+	}
```

Comment:
Couldn't this be removed and instead call `diskTraitsFromEMMCDevice` after the call to `EnsureVolumeCompatibility` in `DiskTraitsFromDeviceAndValidate`? As now that function calls `ensureVolumeEMMCCompatibility` if needed.